pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -698,7 +773,6 @@ cdef class StringHashTable(HashTable):
         v = get_c_string(key)
 
         k = kh_put_str(self.table, v, &ret)
-        self.table.keys[k] = v
```

Comment:
why isnt this needed?

Reply:
`kh_put_str` already puts v into `keys`. There are some places where the additional `self.table.keys[k] = v` was used and some where it wasn't used. For consistency sake (and because my first failed attempt was to convert `double complex` to `kh_complex128_t` in the kh_xxx_xxx-functions) I deleted the unnecessary ones.

Reply:
OK.  this particular part of the code doesn't get many eyeballs, so if there are cleanups you find along the way, go for it (ideally in separate PR)